orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -153,30 +154,46 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
     }
 
     // supposed to always be a private method used strictly by data and delete write builders
-    private WriteBuilder createContextFunc(Function<Configuration, Context> newCreateContextFunc) {
+    private WriteBuilder createContextFunc(Function<Map<String, String>, Context> newCreateContextFunc) {
       this.createContextFunc = newCreateContextFunc;
       return this;
     }
 
     public <D> FileAppender<D> build() {
       Preconditions.checkNotNull(schema, "Schema is required");
 
+      Configuration conf;
+      if (file instanceof HadoopOutputFile) {
+        conf = ((HadoopOutputFile) file).getConf();
+      } else {
+        conf = new Configuration();
+      }
+
+      for (Map.Entry<String, String> entry : config.entrySet()) {
+        conf.set(entry.getKey(), entry.getValue());
+      }
+
+      // for compatibility
+      if (conf.get(VECTOR_ROW_BATCH_SIZE) != null && config.get(TableProperties.ORC_VECTOR_ROW_BATCH_SIZE) == null) {
+        config.put(TableProperties.ORC_VECTOR_ROW_BATCH_SIZE, conf.get(VECTOR_ROW_BATCH_SIZE));
+      }
```

Comment:
Get the value from the Hadoop conf for compatibility. @openinx

Reply:
Compatibility with what? This was unreleased. I don't think that Iceberg should source configuration from the Hadoop Configuration.

Reply:
Okay, I see. This looks fine to me.

Reply:
This looks good to me.

Reply:
It looks like the old config could also be used in `set` on this builder. Should this support that case as well?

Reply:
Hi Ryan, what does 'old config' mean?

Reply:
I think @rdblue means:  we can also set the `iceberg.orc.vectorbatch.size` key in the iceberg table properties (Because we just `set` this key value into the hadoop configuration in the original version and then read this key from the hadoop configuration.)

We should also be able to read this config from `config` table properties if people just set it as the old approach. 

In summary,  we shouldn't need to change those unit tests config setting ([1], [2], [3]). if we keep the old usage.

[1]. https://github.com/apache/iceberg/pull/4291/files#diff-37a827d65fdb4bbf9ce2247d6839a846e27c371eb0bad58a47a4bc924170e2f7L124-L126
[2]. https://github.com/apache/iceberg/pull/4291/files#diff-8878d128d226b30d2941cd46b0e8c9aa193d752355288a6dc928af00d35bf811L124-L126
[3], https://github.com/apache/iceberg/pull/4291/files#diff-86cc5e259c85e8e36c90347a4623fd49e2e7a79d81ef4d8d3c8527111bfc8335L124-L126

Reply:
thx. I tried to reverted unit test changes. and read it from `config`.